EchoPro/survey.py

```diff
@@ -36,741 +25,457 @@ class Survey:
+            raise FileNotFoundError(f"The following data files do not exist: {missing_data}")
+        
+        ### Data validation and import
+        # Iterate through each file defined/detected in 'data_file_paths'
+        for file in data_file_paths:
+
+            # Parse data layers, associated configuration key tag, and the Excel sheet name associated
+            # with the data
+            data_layers, reference_path, sheet_name = self.search_nested_dict(self.config, file)
+
+            # Reference LAYER_NAME_MAP to update 'self.config' -- this is necessary for both mapping
+            # the data attribute structures and reorganizing the dictionary organization within each
+            # class attribute later on
+            # Pull LAYER_NAME_MAP reference information
+            layer_config = LAYER_NAME_MAP.get(data_layers[0])
+
+            # Determine how deep 'data_layers' is represented within the dictionary
+            # This is required for mapping dataframes with an appropriately named
+            # data layer name appended with '_df' (e.g. 'length' -> 'length_df')
+            data_name_bool = set(data_layers).intersection(layer_config['data'])
+            data_layer_index = data_layers.index(''.join(data_name_bool))
+
+            #pull_nested_dict(CONFIG_MAP, data_layers)
+            # Retrieve/access CONFIG_MAP values defining the expected column names and data types
+            validation_settings = pull_nested_dict(CONFIG_MAP, data_layers[:data_layer_index+1])
+
+            ## ??? TODO: Remove Try-Exception ? Or may need more informative error message construction
+            # Validate column names
+            try:
+                # Open connection with the workbook and specific sheet 
+                # This is useful for not calling the workbook into memory and allows for parsing
+                # only the necessary rows/column names
+                workbook = load_workbook(file, read_only=True)
+                sheet = workbook[sheet_name]
+
+                # Validate that the expected columns are contained within the parsed 
+                # column names of the workbook   
+                if reference_path == "kriging/vario_krig_para":
+                    data_columns = [list(row) for row in zip(*sheet.iter_rows(values_only=True))][0]
+                else:   
+                    data_columns = {col.value for col in sheet[1]}
+
+                # Close connection to the work book
+                workbook.close()
+
+                # Error evaluation and print message (if applicable)
+                if not set(validation_settings.keys()).issubset(set(data_columns)):
+                    missing_columns = set(validation_settings.keys()) - set(data_columns)
+                    raise ValueError(f"Missing columns in the Excel file: {missing_columns}")
+                
+            except Exception as e:
+                print(f"Error reading file '{file}': {e}")
+                continue
```

Comment:
I suggest factoring this part out to a function that checks for column name, to keep the main code flow in `load_survey_data` clean and easy to follow.